internal/ir/builder.go

```diff
@@ -26,12 +26,35 @@ import (
 // TODO(matheus): Decide how to deal with top level expressions on f.expresions.
 func (f *File) Build() {
 	for _, member := range f.Members {
-		if fn, ok := member.(*Function); ok {
-			fn.Build()
+		switch m := member.(type) {
+		case *Function:
+			m.Build()
+		case *Struct:
+			m.Build()
 		}
 	}
 }
 
+// Build builds all function members of a struct s.
+//
+// For every method of a struct a receiver parameter is added. This parameter
+// is named self and added as the first index of method.Signature.Params slice
+func (s *Struct) Build() {
+	for _, method := range s.Methods {
+		method.Build()
+
+		p := &Parameter{
+			node:     node{nil},
```

Comment:
I think that would be better to add a comment here
`// Since this parameter don't exists in source code, the syntax for this parameter is nil.`

Reply:
Done!